soh/soh/Enhancements/randomizer/randomizer_item_tracker.cpp

```diff
@@ -659,16 +664,20 @@ void DrawDungeonItem(ItemTrackerItem item) {
                      ImVec2(iconSize, iconSize), ImVec2(0, 0), ImVec2(1, 1));
     }
 
-    if (ResourceMgr_IsSceneMasterQuest(item.data) && (CHECK_DUNGEON_ITEM(DUNGEON_MAP, item.data) || item.data == SCENE_GERUDO_TRAINING_GROUND || item.data == SCENE_INSIDE_GANONS_CASTLE)) {
-        dungeonColor = IM_COL_PURPLE;
+    if (CheckTracker::IsAreaSpoiled(RandomizerCheckObjects::GetRCAreaBySceneID((SceneID)item.data)) && GameInteractor::IsSaveLoaded()) {
+        if (ResourceMgr_IsSceneMasterQuest(item.data))
+            dungeonColor = IM_COL_PURPLE;
+        else
+            dungeonColor = IM_COL_LIGHT_YELLOW;
     }
 
     if (itemId == ITEM_KEY_SMALL) {
-        DrawItemCount(item);
+        DrawItemCount(item, !CheckTracker::IsAreaSpoiled(RandomizerCheckObjects::GetRCAreaBySceneID((SceneID)item.data)));
 
         ImVec2 p = ImGui::GetCursorScreenPos();
+        int offset = GameInteractor::IsSaveLoaded() ? 16 : 13;
```

Comment:
i know we already had 16 as a magic number before, but now i'm even less sure what these numbers mean

not a blocker, but adding a comment explaining why it should be 16 with the save loaded and 13 without might be nice

Reply:
It's basically a pixel count. For some reason, the alignment is 3 units higher if the save is loaded. Those are just the two that put the text at the right spot.

Reply:
Also, 13 is where everything else is at.